File: src/algorithms/strings/transformation/string-rotation-check/sources/string-rotation-check_test.go
package main

import "testing"

func TestStringRotationCheckValidRotation(t *testing.T) {
	if !stringRotationCheck("waterbottle", "erbottlewat") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckZeroOffset(t *testing.T) {
	if !stringRotationCheck("hello", "hello") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckSingleCharMatch(t *testing.T) {
	if !stringRotationCheck("a", "a") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckSingleCharNoMatch(t *testing.T) {
	if stringRotationCheck("a", "b") {
		t.Error("expected false")
	}
}

func TestStringRotationCheckDifferentLengths(t *testing.T) {
	if stringRotationCheck("abc", "ab") {
		t.Error("expected false")
	}
}

func TestStringRotationCheckBottlewater(t *testing.T) {
	if !stringRotationCheck("waterbottle", "bottlewater") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckNotARotation(t *testing.T) {
	if stringRotationCheck("abcde", "abced") {
		t.Error("expected false")
	}
}

func TestStringRotationCheckRotationByOne(t *testing.T) {
	if !stringRotationCheck("abcde", "bcdea") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckRotationFromEnd(t *testing.T) {
	if !stringRotationCheck("abcde", "eabcd") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckTwoEmptyStrings(t *testing.T) {
	if !stringRotationCheck("", "") {
		t.Error("expected true")
	}
}

func TestStringRotationCheckOneEmptyOneNot(t *testing.T) {
	if stringRotationCheck("abc", "") {
		t.Error("expected false")
	}
}

func TestStringRotationCheckRepeatedCharsNotRotation(t *testing.T) {
	if stringRotationCheck("aabaa", "baaab") {
		t.Error("expected false")
	}
}

func TestStringRotationCheckRepeatedCharsValidRotation(t *testing.T) {
	if !stringRotationCheck("aab", "baa") {
		t.Error("expected true")
	}
}

File: src/algorithms/strings/transformation/string-rotation-check/StringRotationCheck_test.java
/** Correctness tests for the StringRotationCheck algorithm. */
public class StringRotationCheck_test {
    public static void main(String[] args) {
        assert StringRotationCheck.stringRotationCheck("waterbottle", "erbottlewat") == true;
        assert StringRotationCheck.stringRotationCheck("hello", "hello") == true;
        assert StringRotationCheck.stringRotationCheck("a", "a") == true;
        assert StringRotationCheck.stringRotationCheck("a", "b") == false;
        assert StringRotationCheck.stringRotationCheck("abc", "ab") == false;
        assert StringRotationCheck.stringRotationCheck("waterbottle", "bottlewater") == true;
        assert StringRotationCheck.stringRotationCheck("abcde", "abced") == false;
        assert StringRotationCheck.stringRotationCheck("abcde", "bcdea") == true;
        assert StringRotationCheck.stringRotationCheck("abcde", "eabcd") == true;
        assert StringRotationCheck.stringRotationCheck("", "") == true;
        assert StringRotationCheck.stringRotationCheck("abc", "") == false;
        assert StringRotationCheck.stringRotationCheck("aabaa", "baaab") == false;
        assert StringRotationCheck.stringRotationCheck("aab", "baa") == true;
        System.out.println("All tests passed!");
    }
}
